pkg/controller/core.oam.dev/v1alpha2/application/template/template.go

```diff
@@ -0,0 +1,79 @@
+package template
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/defclient"
+)
+
+type manager struct {
+	defclient.DefinitionClient
+}
+
+// GetHanler  get template handler
+func GetHanler(cli defclient.DefinitionClient) Handler {
+	m := &manager{
+		DefinitionClient: cli,
+	}
+	return m.LoadTemplate
+}
+
+// Handler is template handler type
+type Handler func(key string) (string, Kind, error)
+
+// Kind is template kind
+type Kind uint16
```

Comment:
please use annotation(`app.oam.dev/resourceType`) in WorkloadDefinition/TraitDefinition to define kind.

Reply:
workload definition template must be workload kind, and trait definition is the same, what is the meaning of the annotation

Reply:
```
// ResourceTypeTrait mark this K8s Custom Resource is an OAM trait
	ResourceTypeTrait = "TRAIT"
	// ResourceTypeWorkload mark this K8s Custom Resource is an OAM workload
	ResourceTypeWorkload = "WORKLOAD"
```